src/reporting.ts

```diff
@@ -54,26 +55,38 @@ export function createIssueCommentMessage(issue: IssueOccurrence): string {
   const message = createReviewCommentMessage(issue)
   const relativePath = relativizePath(issue.mainEventFilePathname)
 
+
   return `${message}
 ## Issue location
 This issue was discovered outside the diff for this Pull Request. You can find it at:
 [${relativePath}:${issue.mainEventLineNumber}](${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/blob/${process.env.GITHUB_SHA}/${relativePath}#L${issue.mainEventLineNumber})
 `
 }
 
+// naive Z-Table, can be optimized to terminate early
+function zTable(input: string) {
+    let table = new Array<number>(input.length).fill(0)
+    for (let i = 1; i < input.length; i++) {
+        for (let j = 0; j < input.length - i; j ++) {
+            if (input.charAt(j) != input.charAt(i + j)) continue
+            table[i]++
+        }
+    }
+    return table
+}
+
 export function getDiffMap(rawDiff: string): DiffMap {
   console.info('Gathering diffs...')
   const diffMap = new Map()
 
   let path = UNKNOWN_FILE
   for (const line of rawDiff.split('\n')) {
     if (line.startsWith('diff --git')) {
-      // TODO: Handle spaces in path
-      path = `${process.env.GITHUB_WORKSPACE}/${line.split(' ')[2].substring(2)}`
-      if (path === undefined) {
-        path = UNKNOWN_FILE
-      }
-
+      // this should give `diff --git a/path b/path`
+      // to get the file itself, we can use longest string match on `path b/path`
+      // since we definitely know paths are equal, path must be the longest string
+      path = line.substring("diff --git a/".length)
+      path = path.substring(0, Math.max(...zTable(path)))
```

Comment:
basically doing a naive longest string match here to find the path without splitting